registrator.go

```diff
@@ -172,6 +173,10 @@ func main() {
 			go b.Add(msg.ID)
 		case "die":
 			go b.RemoveOnExit(msg.ID)
+		case "kill":
+			if *deregisterOnStop {
+				go b.RemoveOnExit(msg.ID)
+			}
 		}
```

Comment:
This code will cause unexpected results with a command like `docker kill -s=SIGHUP some-container`. `docker kill` will emit a kill event regardless of the signal sent. 

Here is an issue regarding the events documentation docker/docker#21331.